File: app/src/main/java/com/speedroid/macroid/ui/activity/ModeActivity.java
package com.speedroid.macroid.ui.activity;

import static com.speedroid.macroid.Configs.DIALOG_TYPE_SELECT_MODE;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.speedroid.macroid.ui.fragment.dialog.DefaultDialogFragment;


public class ModeActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        DefaultDialogFragment defaultDialogFragment = new DefaultDialogFragment(DIALOG_TYPE_SELECT_MODE);
        defaultDialogFragment.show(getSupportFragmentManager(), Integer.toString(DIALOG_TYPE_SELECT_MODE));
    }
}

File: app/src/main/java/com/speedroid/macroid/ui/activity/ModeActivity.kt
package com.speedroid.macroid.ui.activity

import android.annotation.SuppressLint
import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.speedroid.macroid.Configs.Companion.DIALOG_TYPE_MODE
import com.speedroid.macroid.ui.fragment.dialog.RecyclerDialogFragment

class ModeActivity : AppCompatActivity() {
    companion object {
        @SuppressLint("StaticFieldLeak")
        lateinit var preservedContext: Context
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // initialize application context
        preservedContext = this

        RecyclerDialogFragment(DIALOG_TYPE_MODE).show(supportFragmentManager, DIALOG_TYPE_MODE.toString())
    }
}